applications/ContactStructuralMechanicsApplication/custom_utilities/contact_utilities.h

```diff
@@ -78,6 +91,95 @@ class ContactUtilities
     ///@name Operations
     ///@{
     
+    /**
+     * @brief This function computes the relative size of the mesh
+     * @param rThisModelPart The modelpart to compute
+     */
+
+    static inline double CalculateRelativeSizeMesh(ModelPart& rThisModelPart)
+    {
+        return CalculateMaxNodalH(rThisModelPart)/CalculateMinimalNodalH(rThisModelPart);
+    }
+
+    /**
+     * @brief This method computes the maximal nodal H
+     * @param rThisModelPart The modelpart to compute
+     */
+    static inline double CalculateMaxNodalH(ModelPart& rThisModelPart)
+    {
+        // We iterate over the nodes
+        NodesArrayType& nodes_array = rThisModelPart.Nodes();
+
+        // Creating a buffer for parallel vector fill
+        const int num_threads = OpenMPUtils::GetNumThreads();
+        std::vector<double> max_vector(num_threads, 0.0);
+        double nodal_h;
+        #pragma omp parallel for private(nodal_h)
+        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
+            auto it_node = nodes_array.begin() + i;
+            KRATOS_DEBUG_ERROR_IF_NOT(it_node->SolutionStepsDataHas(NODAL_H)) << "ERROR:: NODAL_H not added" << std::endl;
+            nodal_h = it_node->FastGetSolutionStepValue(NODAL_H);
+
+            const int id = OpenMPUtils::ThisThread();
+
+            if (nodal_h > max_vector[id])
+                max_vector[id] = nodal_h;
+        }
+
+        return *std::max_element(max_vector.begin(), max_vector.end());
+    }
+
+    /**
+     * @brief This method computes the mean nodal H
+     * @param rThisModelPart The modelpart to compute
+     */
+    static inline double CalculateMeanNodalH(ModelPart& rThisModelPart)
+    {
+        // We iterate over the nodes
+        NodesArrayType& nodes_array = rThisModelPart.Nodes();
+
+        double nodal_h;
+        double sum_nodal_h = 0.0;
+
+        #pragma omp parallel for  private(nodal_h) reduction(+:sum_nodal_h)
+        for(int i = 0; i < static_cast<int>(nodes_array.size()); ++i) {
+            auto it_node = nodes_array.begin() + i;
+            KRATOS_DEBUG_ERROR_IF_NOT(it_node->SolutionStepsDataHas(NODAL_H)) << "ERROR:: NODAL_H not added" << std::endl;
+            nodal_h = it_node->FastGetSolutionStepValue(NODAL_H);
```

Comment:
why do you need this extra step?
=>How abt  `sum_nodal_h += it_node->FastGetSolutionStepValue(NODAL_H);`?

Reply:
Ok